biometry/src/androidMain/kotlin/dev/icerock/moko/biometry/BiometryAuthenticator.kt

```diff
@@ -132,26 +132,26 @@ actual class BiometryAuthenticator actual constructor() {
 
             biometricPrompt = BiometricPrompt(this, executor,
                 object : BiometricPrompt.AuthenticationCallback() {
-                    override fun onAuthenticationError(errorCode: Int,
-                                                       errString: CharSequence) {
+                    override fun onAuthenticationError(
+                        errorCode: Int,
+                        errString: CharSequence
+                    ) {
                         super.onAuthenticationError(errorCode, errString)
-                        if (errorCode == BiometricConstants.ERROR_NEGATIVE_BUTTON) {
-                            callback.invoke(Result.failure(Exception(errorCode.toString())))
+                        if (errorCode == BiometricPrompt.ERROR_NEGATIVE_BUTTON ||
+                            errorCode == BiometricPrompt.ERROR_USER_CANCELED
+                        ) {
+                            callback.invoke(Result.success(false))
                         } else {
                             callback.invoke(Result.failure(Exception(errString.toString())))
                         }
                     }
 
                     override fun onAuthenticationSucceeded(
-                        result: BiometricPrompt.AuthenticationResult) {
+                        result: BiometricPrompt.AuthenticationResult
+                    ) {
                         super.onAuthenticationSucceeded(result)
                         callback.invoke(Result.success(true))
                     }
-
-                    override fun onAuthenticationFailed() {
```

Comment:
why this removed?

Reply:
At the documentation: 
> onAuthenticationFailed()  Called when a biometric is valid but not recognized. 

Is it the expected behavior of `checkBiometricAuthentication` when function terminates and returns the result when a fingerprint is incorrect? As if the user has only one attempt.

Reply:
if auth dialog will be dissmissed after `onAuthenticationFailed` then we should call callback. did you check this case?

Reply:
After dismissing a dialog callback `onAuthenticationError` is called, not `onAuthenticationFailed`. Callback `onAuthenticationFailed` is called when a fingerprint is incorrect, for example, when user put wrong finger. After incorrect fingerprint the biometric dialog doesn't dismiss automatically.

Reply:
In this case you are right - we should not pass result to coroutines-test at failed call